plugins/modules/ndo_mac_sec_policy.py

```diff
@@ -0,0 +1,440 @@
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_test_template
+    state: query
+  register: query_all
+
+- name: Query a MACSec Policy with mac_sec_policy UUID
+  cisco.mso.ndo_mac_sec_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_test_template
+    mac_sec_policy_uuid: ansible_test_mac_sec_policy_uuid
+    state: query
+  register: query_uuid
+
+- name: Delete a MACSec Policy with name
+  cisco.mso.ndo_mac_sec_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_test_template
+    mac_sec_policy: ansible_test_mac_sec_policy
+    state: absent
+
+- name: Delete a MACSec Policy with UUID
+  cisco.mso.ndo_mac_sec_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_test_template
+    mac_sec_policy_uuid: ansible_test_mac_sec_policy_uuid
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import MSOModule, mso_argument_spec
+from ansible_collections.cisco.mso.plugins.module_utils.template import MSOTemplate, KVPair
+from ansible_collections.cisco.mso.plugins.module_utils.constants import NDO_CIPHER_SUITE_MAP, NDO_SECURITY_POLICY_MAP
+import copy
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        dict(
+            template=dict(type="str", required=True),
+            mac_sec_policy=dict(type="str", aliases=["name"]),
+            mac_sec_policy_uuid=dict(type="str", aliases=["uuid"]),
+            description=dict(type="str"),
+            admin_state=dict(type="str", choices=["enabled", "disabled"]),
+            interface_type=dict(type="str", choices=["fabric", "access"], default="fabric"),
+            cipher_suite=dict(type="str", choices=list(NDO_CIPHER_SUITE_MAP.keys())),
+            window_size=dict(type="int"),
+            security_policy=dict(type="str", choices=list(NDO_SECURITY_POLICY_MAP.keys())),
+            sak_expiry_time=dict(type="int"),
+            confidentiality_offset=dict(type="int", choices=[0, 30, 50]),
+            key_server_priority=dict(type="int"),
+            mac_sec_key=dict(
+                type="list",
+                elements="dict",
+                options=dict(
+                    key_name=dict(type="str", required=True),
+                    psk=dict(type="str", required=True, no_log=True),
+                    start_time=dict(type="str"),
+                    end_time=dict(type="str"),
+                ),
+                no_log=False,
+            ),
+            state=dict(type="str", choices=["absent", "query", "present"], default="query"),
+        )
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "present", ["mac_sec_policy"]],
```

Comment:
```suggestion
            ["state", "present", ["mac_sec_policy", "mac_sec_policy_uuid"], True],
```

We can allow users to update using only UUID without a `mac_sec_policy`.

Reply:
changed